core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -153,8 +153,13 @@ class KryoSerializer(conf: SparkConf)
     kryo.register(classOf[SerializableJobConf], new KryoJavaSerializer())
     kryo.register(classOf[PythonBroadcast], new KryoJavaSerializer())
 
-    kryo.register(classOf[GenericRecord], new GenericAvroSerializer(avroSchemas))
-    kryo.register(classOf[GenericData.Record], new GenericAvroSerializer(avroSchemas))
+    def registerAvro[T >: Null <: GenericContainer](implicit ct: ClassTag[T]) =
+      kryo.register(ct.runtimeClass, new GenericAvroSerializer[T](avroSchemas))
+    registerAvro[GenericRecord]
+    registerAvro[GenericData.Record]
+    registerAvro[GenericData.Array[_]]
+    registerAvro[GenericData.EnumSymbol]
+    registerAvro[GenericData.Fixed]
```

Comment:
I think for specific records we will need to use a different codepath inside Avro for serde viz. `SpecificDatum(Reader|Writer)` instead of `GenericDatum(Reader|Writer)` which is currently how `GenericAvroSerializer` is implemented. Also I think we will need to register each concrete SpecificRecord class for Kryo to work correctly. Thoughts?

Reply:
To clarify, can we add that a comment ? :-)

Reply:
Ahh! Done